File: data/recomposed_projects/llama-3-70b-instruct/body/commons-validator/evosuite-test/src/test/org/apache/commons/validator/EmailValidator_ESTest.py
from __future__ import annotations
import time
import re
import os
import typing
from typing import *
import unittest
import pytest
import io
import unittest
from src.main.org.apache.commons.validator.EmailValidator import *

# from src.test.org.apache.commons.validator.EmailValidator_ESTest_scaffolding import *
# from src.main.org.evosuite.runtime.EvoAssertions import *
# from src.main.org.evosuite.runtime.EvoRunnerParameters import *
from src.main.org.apache.commons.validator.routines.DomainValidator import *
from src.main.org.apache.commons.validator.routines.EmailValidator import *

# from src.test.org.apache.commons.validator.routines.EmailValidator_ESTest_scaffolding import *


class EmailValidator_ESTest(unittest.TestCase):

    def test16(self) -> None:

        email_validator = EmailValidator()

        try:
            email_validator._isValidDomain("[]")
            self.fail("Expecting exception: RuntimeError")
        except RuntimeError as e:
            self.verifyException("java.util.regex.Matcher", e)

    def test15(self) -> None:

        emailValidator0 = EmailValidator.getInstance()
        boolean0 = emailValidator0.isValid("5#_3F.z 5")
        self.assertFalse(boolean0)

    def test14(self) -> None:
        emailValidator0 = EmailValidator.getInstance()
        boolean0 = emailValidator0._isValidUser(
            "m" + "!" + ";" + '"' + "/" + "YWY" + "/" + "!" + "`" + "2" + "=" + "?"
        )
        self.assertFalse(boolean0)

    def test13(self) -> None:

        email_validator0 = EmailValidator.getInstance()

        with pytest.raises(ArrayIndexOutOfBoundsException):
            email_validator0.isValidDomain("g2")

    def test12(self) -> None:

        email_validator0 = EmailValidator.getInstance()
        boolean0 = email_validator0._isValidDomain("b[]")
        self.assertFalse(boolean0)

    def test11(self) -> None:

        email_validator0 = EmailValidator()
        # Undeclared exception
        try:
            email_validator0.isValidIpAddress("`l")
            self.fail("Expecting exception: RuntimeError")

        except RuntimeError as e:
            # No match found
            self.verifyException("java.util.regex.Matcher", e)

    def test10(self) -> None:

        email_validator0 = EmailValidator.getInstance()
        # Undeclared exception handling
        try:
            email_validator0._isValidDomain("?zP-^cj.U$?rt")
        except Exception as e:
            print(f"Exception occurred: {e}")

    def test09(self) -> None:

        emailValidator0 = EmailValidator.getInstance()
        string0 = emailValidator0._stripComments("u|R]lU |#8E&u`-V}[9")
        self.assertEqual("u|R]lU |#8E&u`-V}[9", string0)

    def test08(self) -> None:

        email_validator = EmailValidator()

        try:
            email_validator._isValidDomain(None)
            self.fail("Expecting exception: RuntimeError")

        except TypeError as e:
            # no message in exception (getMessage() returned null)
            pass

    def test07(self) -> None:

        email_validator0 = EmailValidator()
        # Undeclared exception in Java code
        try:
            email_validator0.isValidIpAddress(None)
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            # no message in exception (getMessage() returned null)
            pass

    def test06(self) -> None:

        email_validator0 = EmailValidator()
        # Undeclared exception in Java code
        try:
            email_validator0._isValidSymbolicDomain("cool")
            self.fail("Expecting exception: ArrayIndexOutOfBoundsException")

        except IndexError:
            # no message in exception (getMessage() returned null)
            pass

    def test05(self) -> None:

        email_validator0 = EmailValidator.getInstance()
        # Undeclared exception in Java code
        try:
            email_validator0._isValidSymbolicDomain(None)
            self.fail("Expecting exception: RuntimeError")

        except Exception as e:
            # no message in exception (getMessage() returned null)
            pass

    def test04(self) -> None:

        email_validator0 = EmailValidator()
        # Undeclared exception handling is not necessary in Python.
        email_validator0._isValidSymbolicDomain('"<Ja4p.bwv{&&v3$')

    def test03(self) -> None:
        email_validator0 = EmailValidator()
        try:
            email_validator0._isValidUser(None)
            self.fail("Expecting exception: RuntimeError")
        except TypeError as e:
            # no message in exception (getMessage() returned null)
            pass

    def test02(self) -> None:

        emailValidator0 = EmailValidator()
        # Undeclared exception in Java code
        try:
            emailValidator0._stripComments(None)
            self.fail("Expecting exception: RuntimeError")

        except TypeError as e:
            # no message in exception (getMessage() returned null)
            pass

    def test01(self) -> None:
        email_validator0 = EmailValidator()
        boolean0 = email_validator0._isValidUser("'_7K=")
        self.assertTrue(boolean0)

    def test00(self) -> None:

        emailValidator0 = EmailValidator.getInstance()
        string0 = emailValidator0._stripComments("")
        self.assertEqual("", string0)

File: java_projects/cleaned_final_projects_evosuite/commons-validator/evosuite-tests/org/apache/commons/validator/EmailValidator_ESTest.java


/*
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */



package org.apache.commons.validator;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.apache.commons.validator.EmailValidator;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true) 
public class EmailValidator_ESTest extends EmailValidator_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      String string0 = emailValidator0.stripComments("");
      assertEquals("", string0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      boolean boolean0 = emailValidator0.isValidUser("'_7K=");
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      try { 
        emailValidator0.stripComments((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      try { 
        emailValidator0.isValidUser((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      emailValidator0.isValidSymbolicDomain("\"<Ja4p.bwv{&&v3$");
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      // Undeclared exception!
      try { 
        emailValidator0.isValidSymbolicDomain((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      try { 
        emailValidator0.isValidSymbolicDomain("cool");
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      try { 
        emailValidator0.isValidIpAddress((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      try { 
        emailValidator0.isValidDomain((String) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      String string0 = emailValidator0.stripComments("u|R]lU |#8E&u`-V}[9");
      assertEquals("u|R]lU |#8E&u`-V}[9", string0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      // Undeclared exception!
      emailValidator0.isValidDomain("?zP-^cj.U$?rt");
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      try { 
        emailValidator0.isValidIpAddress("`l");
        fail("Expecting exception: IllegalStateException");
      
      } catch(IllegalStateException e) {
         //
         // No match found
         //
         verifyException("java.util.regex.Matcher", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      boolean boolean0 = emailValidator0.isValidDomain("b[]");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      // Undeclared exception!
      try { 
        emailValidator0.isValidDomain("g2");
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      boolean boolean0 = emailValidator0.isValidUser("m!;\"[/YWY/!`2=?");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      EmailValidator emailValidator0 = EmailValidator.getInstance();
      boolean boolean0 = emailValidator0.isValid("5#_3F.z 5");
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      EmailValidator emailValidator0 = new EmailValidator();
      // Undeclared exception!
      try { 
        emailValidator0.isValidDomain("[]");
       //  fail("Expecting exception: IllegalStateException");
       // Unstable assertion
      } catch(IllegalStateException e) {
         //
         // No match found
         //
         verifyException("java.util.regex.Matcher", e);
      }
  }
}
